packages/core/backend/sirius-components-core/src/main/java/org/eclipse/sirius/components/core/api/IObjectServiceDelegate.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.core.api;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Interface of the delegation service interacting with domain objects.
+ *
+ * @author arichard
+ */
+public interface IObjectServiceDelegate {
+
+    boolean canHandle(IEditingContext editingContext, Object object);
```

Comment:
I would not give access to the `IEditingContext` in this method because most methods here do not give us the ability to leverage the editing context and if some application needs to find out the editing context from the object it can be done in application-specific code.

Reply:
After a look at the rest of the code, I'm even more convinced since you have to give `null` most of the time since you can't do anything else. We will never be able to give the editing context in most use cases and applications can retrieve it by themselves if they want so no need for such variable.